cli.go

```diff
@@ -0,0 +1,580 @@
+				cli.StringFlag{
+					Name:        "tenant-token, T",
+					Destination: &runOptions.setupOptions.tenantToken,
+					Usage:       "Hosted Mender tenanant `token`"},
+				cli.IntFlag{
+					Name:        "inventory-poll, I",
+					Destination: &runOptions.setupOptions.invPollInterval,
+					Usage:       "Inventory poll interval in `sec`onds."},
+				cli.IntFlag{
+					Name:        "retry-poll, R",
+					Destination: &runOptions.setupOptions.retryPollInterval,
+					Usage:       "Retry poll interval in `sec`onds."},
+				cli.IntFlag{
+					Name:        "update-poll, P",
+					Destination: &runOptions.setupOptions.updatePollInterval,
+					Usage:       "Update poll interval in `sec`onds."},
+				cli.BoolFlag{
+					Name:        "hosted-mender, H",
+					Destination: &runOptions.setupOptions.hostedMender,
+					Usage:       "Setup device towards Hosted Mender."},
+				cli.BoolFlag{
+					Name:        "demo, D",
+					Destination: &runOptions.setupOptions.demo,
+					Usage:       "Use demo configuration."},
+				cli.BoolFlag{
+					Name:  "run-daemon, r",
+					Usage: "Run daemon after setup."},
+			},
+		},
+		{
+			Name: "show-artifact",
+			Usage: "Print the current artifact name to the " +
+				"command line and exit.",
+			Action: runOptions.handleCLIOptions,
+		},
+	}
+	app.Flags = []cli.Flag{
+		cli.StringFlag{
+			Name:        "config, c",
+			Usage:       "Configuration `FILE` path.",
+			Value:       defaultConfFile,
+			Destination: &runOptions.config},
+		cli.StringFlag{
+			Name:        "fallback-config, C",
+			Usage:       "Fallback configuration `FILE` path.",
+			Value:       defaultFallbackConfFile,
+			Destination: &runOptions.fallbackConfig},
+		cli.StringFlag{
+			Name:        "data, d",
+			Usage:       "Mender state data `DIR`ECTORY path.",
+			Value:       defaultDataStore,
+			Destination: &runOptions.dataStore},
+		cli.StringFlag{
+			Name:        "log-file, L",
+			Usage:       "`FILE` to log to.",
+			Destination: &runOptions.logOptions.logFile},
+		cli.StringFlag{
+			Name:        "log-level, l",
+			Usage:       "Set logging `level`.",
+			Value:       "info",
+			Destination: &runOptions.logOptions.logLevel},
+		cli.StringFlag{
+			Name: "log-modules, m",
+			Usage: "`LIST` of logging modules (levels) to " +
+				"include in the output.",
+			Destination: &runOptions.logOptions.logModules},
+		cli.StringFlag{
+			Name:        "trusted-certs, E",
+			Usage:       "Trusted server certificates `FILE` path.",
+			Destination: &runOptions.Config.ServerCert},
+		cli.BoolFlag{
+			Name:        "forcebootstrap, F",
+			Usage:       "Force bootstrap.",
+			Destination: &runOptions.bootstrapForce},
+		cli.BoolFlag{
+			Name:        "no-syslog, S",
+			Usage:       "Disable logging to syslog.",
+			Destination: &runOptions.logOptions.noSyslog},
+		cli.BoolFlag{
+			Name:        "skipverify, k",
+			Usage:       "Skip certificate verification.",
+			Destination: &runOptions.Config.NoVerify},
+	}
```

Comment:
Again, I think I would skip most of the short forms here also.

It is a matter to be where possible consistent across our products. Like -k in mender-artifact has relation with keys, here is about skipping something...

My criteria would be: for anything looks like is going to be used a lot manually (for instance log level), enable the short forms. For the rest use only the long ones...

Reply:
I would certainly recommend using the long option in any application which may possibly read by others (e.g. scripts), on the other hand I like having a short "lazy" way to provide flags. Some of the flag choices were chosen on based of other unix applications such as `-k` and `-4` from `curl`. We don't have that large repertoire of flags that I think it is necessary to exclude the short option, however if you strongly disagree I would like you to be more specific about which flags should be included/excluded/changed.

Reply:
So here's what I've done:
For the setup flags I've kept the short option for `config `data` and `server-cert` which are all shared with the global options.
For the global options I've removed short option for `no-syslog` and `skipverify`.

Reply:
I had no strong opinion on this. But thanks for your effort in simplifying it. I will review once again but now I have to learn about `go generator` tool... :)